File: app/kotlin/puyo-puyo-tdd/src/commonMain/kotlin/PuyoColor.kt
package com.example.puyopuyo

import androidx.compose.ui.graphics.Color

object PuyoColor {
    val Empty = Color.White
    val Red = Color(0xFFFF4444) // 赤ぷよ (type = 1)
    val Blue = Color(0xFF4444FF) // 青ぷよ (type = 2)
    val Green = Color(0xFF44FF44) // 緑ぷよ (type = 3)
    val Yellow = Color(0xFFFFFF44) // 黄ぷよ (type = 4)

    fun getColor(puyoType: Int): Color {
        return when (puyoType) {
            1 -> Red
            2 -> Blue
            3 -> Green
            4 -> Yellow
            else -> Empty
        }
    }
}

File: app/typescript-5/puyo-puyo-game/src/models/PuyoColor.ts
export enum PuyoColor {
  EMPTY = 0,
  RED = 1,
  BLUE = 2,
  GREEN = 3,
  YELLOW = 4,
}

interface ColorStyle {
  backgroundColor: string
}

export const getPuyoColorStyle = (color: PuyoColor): ColorStyle => {
  switch (color) {
    case PuyoColor.RED:
      return { backgroundColor: '#ff4444' }
    case PuyoColor.BLUE:
      return { backgroundColor: '#4444ff' }
    case PuyoColor.GREEN:
      return { backgroundColor: '#44ff44' }
    case PuyoColor.YELLOW:
      return { backgroundColor: '#ffff44' }
    case PuyoColor.EMPTY:
    default:
      return { backgroundColor: 'transparent' }
  }
}
